app/controllers/comments_controller.rb

```diff
@@ -1,15 +1,21 @@
 class CommentsController < ApplicationController
   def create
-    @post_id = params[:post_id]
-    @post = Post.find(params[:post_id])
-    @user = current_user
-    @comments = current_user.comments.create(:post_id => @post_id, :user_id => @user.id, :body => comment_params[:body])
-    if @comments.save
-      flash[:notice] = "Comment was saved."
-      redirect_to url_for(:action=>"show", :controller=>"posts", :format=>"26", :id=>@post.id , :topic_id=>@post.topic_id)
+    if current_user == nil
+      flash[:error] = "Log in first if you want to create a comment"
+      redirect_to '/users/sign_in'      
     else
-      flash[:error] = "There was an error saving the post. Please try again."
-      redirect_to '/topics'
+      puts "--- current_user = #{current_user} ---"
+      @post_id = params[:post_id]
+      @post = Post.find(params[:post_id])
+      @user = current_user
+        @comments = current_user.comments.create(:post_id => @post_id, :user_id => @user.id, :body => comment_params[:body])
```

Comment:
Hi Colby

I agree with all your comments. I incorporated the changes into a new
commit and updated the assignment submission.
Thank you for checking my work.

On Tue, Mar 24, 2015 at 7:37 PM, Colby Aley <notifications@github.com>
wrote:

> In app/controllers/comments_controller.rb
> <https://github.com/stephenwlott/bloccit/pull/1#discussion_r27091283>:
>
> >      else
> > -      flash[:error] = "There was an error saving the post. Please try again."
> > -      redirect_to '/topics'
> > +      puts "--- current_user = #{current_user} ---"
> > +      @post_id = params[:post_id]
> > +      @post = Post.find(params[:post_id])
> > +      @user = current_user
> > +        @comments = current_user.comments.create(:post_id => @post_id, :user_id => @user.id, :body => comment_params[:body])
>
> instead of setting and using @post_id, just use @post.id
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stephenwlott/bloccit/pull/1/files#r27091283>.
>